lint/linter.py

```diff
@@ -1628,27 +1651,28 @@ def split_match(self, match):
         and return them.
 
         """
+        match_dict = MATCH_DICT.copy()
 
-        if match:
-            items = {'line': None, 'col': None, 'error': None, 'warning': None, 'message': '', 'near': None}
-            items.update(match.groupdict())
-            line, col, error, warning, message, near = [
-                items[k] for k in ('line', 'col', 'error', 'warning', 'message', 'near')
-            ]
+        if not match:
+            persist.debug('No match for regex: {}'.format(self.regex.pattern))
+        else:
+            match_dict.update(match.groupdict())
+            match_dict["match"] = match
 
-            if line is not None:
-                line = int(line) - self.line_col_base[0]
+            # normalize line and col if necessary
+            line = match_dict["line"]
+            if line:
+                match_dict["line"] = int(line) - self.line_col_base[0]
 
-            if col is not None:
+            col = match_dict["col"]
+            if col:
                 if col.isdigit():
                     col = int(col) - self.line_col_base[1]
                 else:
                     col = len(col)
+                match_dict["col"] = col
 
-            return match, line, col, error, warning, message, near
-        else:
-            persist.debug('No match for regex: {}'.format(self.regex.pattern))
-            return match, None, None, None, None, '', None
+        return LintMatch(*match_dict.values())
```

Comment:
Could also use `LintMatch(**match_dict)` here. It's an ordered dict anyway, but this seems more explicit to me (despite having fewer characters).

Reply:
Thanks. Didn't know that one!

Reply:
Agreed.